### PR TITLE
[PROTOCOL-22] Add Setter Function For Oracle Price Reference

### PR DESCRIPTION
**Pull request by salazarguille**
It includes:

- A setter function to update the oracle reference.
- New unit tests.

image

### CODE REVIEW

File: contracts/base/Base.sol [hunk 1]
```diff
@@ -67,6 +67,16 @@ contract Base is Initializable, ReentrancyGuard {
...
+    /**
+        @notice Checks whether a given address is allowed (see Settings#hasPauserRole function) or not.
+        @dev It throws a require error if address is not allowed.
+        @param anAddress account to test.
+     */
+    modifier whenAllowed(address anAddress) {
+        require(_isAllowed(anAddress), "ADDRESS_ISNT_ALLOWED");
+        _;
+    }
+
```
Comment (passabilities) [inline contracts/base/Base.sol:79]: seems redundant to have an additional modifier and function that do the same thing as preexisting modifiers.
See other comments as well.

File: contracts/base/Base.sol [hunk 2]
```diff
@@ -103,5 +113,14 @@ contract Base is Initializable, ReentrancyGuard {
...
 
+    /**
+        @notice Tests whether an account has the pauser role.
+        @param anAddress account to test.
+        @return true if account has the pauser role. Otherwise it returns false.
+     */
+    function _isAllowed(address anAddress) internal view returns (bool) {
+        return settings.hasPauserRole(anAddress);
+    }
+
```
Comment (passabilities) [inline contracts/base/Base.sol:124]: seems redundant

File: contracts/base/Base.sol (hunk 1 above)
Comment (salazarguille) [reply contracts/base/Base.sol:79]: Fixed

File: contracts/base/Base.sol (hunk 2 above)
Comment (salazarguille) [reply contracts/base/Base.sol:124]: Fixed

**Pull request merged**